gp-nested-forms/gpnf-child-products-in-parent-order-summary.php

```diff
@@ -40,6 +40,19 @@
 				$_child_products    = array();
 				foreach ( $child_product_info['products'] as $child_field_id => $child_product ) {
 					$child_product['name'] = "{$product_info['products'][ $field->id ]['name']} — {$child_product['name']}";
+
+					// If Nested Form fields have Live Merge Tags, process those.
+					if ( strpos( $child_product['name'], '@{' ) !== false && class_exists( 'GP_Populate_Anything_Live_Merge_Tags' ) ) {
+						$gp_lmt         = GP_Populate_Anything_Live_Merge_Tags::get_instance();
+						$nested_form_id = $field->formId;
+						$nested_form    = GFAPI::get_form( $nested_form_id );
+						$matches        = array();
+						preg_match( $gp_lmt->live_merge_tag_regex, $child_product['name'], $matches );
+
+						$merge_tag_match_value = GFCommon::replace_variables( $matches[1], $nested_form, $child_entry, false, false, false );
+						$child_product['name'] = str_replace( $matches[0], $merge_tag_match_value, $child_product['name'] );
+					}
+
```

Comment:
```suggestion
					// If Nested Form fields have Live Merge Tags, process those.
					if ( method_exists( 'GP_Populate_Anything_Live_Merge_Tags', 'has_live_merge_tag' ) ) {
						$gppa_lmt         = GP_Populate_Anything_Live_Merge_Tags::get_instance();
				
						if ( $gppa_lmt->has_live_merge_tag( $child_product['name'] ) ) {
							$child_product['name'] = $gppa_lmt->replace_live_merge_tags_static( $child_product['name'], GFAPI::get_form( $field->formId ), $child_entry );
						}
					}
```

Reply:
This is untested, but we have some nice methods built in that might be able to simplify this.

Reply:
The simplified logic does make sense, but ain't work. I am confused about 1 particular method: 

`public function get_live_merge_tag_value( $merge_tag, $form, $entry_values = null ) {` ([L831](https://github.com/gravitywiz/gp-populate-anything/blob/master/includes/class-live-merge-tags.php#L831))

what does the 3rd parameter `$entry_values` imply? Is it the `$entry` object or something else? This method gets called in the call stack via `replace_live_merge_tags_static` or `replace_live_merge_tags`

Reply:
Ah, my bad. I forgot to populate the LMT whitelist (it's for security, so we don't just parse arbitrary Live Merge Tags entered by users) and also I provided the wrong form param.

`$entry_values` is _usually_ interchangeable with `$field_values` or `$entry` in the scope of GPPA unless something is explicitly being done to/with the entry.

The following should work:

```php
					// If Nested Form fields have Live Merge Tags, process those.
					if ( method_exists( 'GP_Populate_Anything_Live_Merge_Tags', 'has_live_merge_tag' ) ) {
						$gppa_lmt = GP_Populate_Anything_Live_Merge_Tags::get_instance();

						if ( $gppa_lmt->has_live_merge_tag( $child_product['name'] ) ) {
							$gppa_lmt->populate_lmt_whitelist( $child_form );
							$child_product['name'] = $gppa_lmt->replace_live_merge_tags_static( $child_product['name'], $child_form, $child_entry );
						}
					}
```

Reply:
@claygriffiths Form param was fine, `$field->formId` points to the correct nested form. I had it on my snippet as well. Infact `$child_form` is not pointing correctly.

Adding the LMT whitelist to the current nested form, but that still doesn't resolve the issue. The manual approach of calling GFCommon::replace_variables is the only one that I've working thus far.

Out of curiosity, we do have a filter added  (L81):
`add_filter( 'gpnf_all_entries_nested_entry_markup', array( $this, 'replace_live_merge_tags_gpnf_all_entries' ), 10, 5 );`

The implementation is 

```
	function replace_live_merge_tags_gpnf_all_entries( $markup, $nested_form_field, $nested_form, $entry, $args ) {
		$this->populate_lmt_whitelist( $nested_form );
		return $this->replace_live_merge_tags_static( $markup, $nested_form, $entry );
	}
```

On debugging, I can confirm that the filter does run, and nested form and entry values are correctly populated. It's again the call to `replace_live_merge_tags_static` that does not fetch the results. I think if we fix the issue here, we will not need to make updates to the snippet?

Reply:
@saifsultanc I sent a Loom to you regarding this. Thanks for continuing to dig in!